### DISCUSSION TITLE
如何自定一个 `EditorDocumentModel`

### DISCUSSION THREAD

**Discussion by l1shen** [opensumi/core#2553] [Q&A]
我们使用自定义资源和编辑器富组件（React 组件）的方式来渲染特定资源。用户可以通过直接通过操作这个编辑器富组件来修改资源
然后使用 `IEditorDocumentModel.updateContent(JSON.stringify(data))` 方法将数据同步回文件。
这个过程存在以下问题：

1. 用户每次操作都会触发数据序列化并调用 updateContent 方法，如果数据变大之后频繁进行 stringify 可能是个没必要性能消耗
2. 如果不使用自带的这个 `EditorDocumentModel` 有没有方法自己实现一个 `IEditorDocumentModel`

**Comment by erha19** [comment]
@l1shen 有比较完整伪代码吗？看一下具体用法，另外，如果完全脱离 Editor 的文档管理，你也可以基于 FileService 去做文件的存取。

**Reply by l1shen** [reply]
可能我这里的用法不是很对，我描述一下具体的需求，
就是我们想要根据一个文件自己去渲染一套 UI
但是又想要这个渲染出来的 UI 完全走 editor 的逻辑，比如可以同时分栏打开这个文件，一边修改，另外一边会同步这个修改，
然后还会走编辑器完整的自动保存逻辑等

你这里会建议通过哪种方式实现这个需求呢

**Reply by erha19** [reply]
@l1shen 可以参考这个案例去实现 https://github.com/opensumi/opensumi-module-samples/tree/main/modules/use-antd， 非文本编辑器，完全走 Edtior 逻辑是做不到的，你需要自行处理文件的更新同步能力。可以借助文件事件及编辑器事件去做。
